server/datastore/mysql/software.go

```diff
@@ -4250,3 +4250,96 @@ WHERE hvsi.host_id = ? AND st.id IN (?)
 	}
 	return nil
 }
+
+func (ds *Datastore) NewSoftwareCategory(ctx context.Context, name string) (*fleet.SoftwareCategory, error) {
+	stmt := `INSERT INTO software_categories (name) VALUES (?)`
+	res, err := ds.writer(ctx).ExecContext(ctx, stmt, name)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "new software category")
+	}
+
+	r, _ := res.LastInsertId()
+	id := uint(r) //nolint:gosec // dismiss G115
+	return &fleet.SoftwareCategory{Name: name, ID: id}, nil
+}
+
+func (ds *Datastore) GetSoftwareCategoryIDs(ctx context.Context, names []string) ([]uint, error) {
+	if len(names) == 0 {
+		return []uint{}, nil
+	}
+
+	stmt := `SELECT id FROM software_categories WHERE name IN (?)`
+	stmt, args, err := sqlx.In(stmt, names)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "sqlx.In for get software category ids")
+	}
+
+	var ids []uint
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &ids, stmt, args...); err != nil {
+		if !errors.Is(err, sql.ErrNoRows) {
+			return nil, ctxerr.Wrap(ctx, err, "get software category ids")
+		}
+	}
+
+	return ids, nil
+}
+
+func (ds *Datastore) GetCategoriesForSoftwareTitles(ctx context.Context, softwareTitleIDs []uint, teamID *uint) (map[uint][]string, error) {
+	if len(softwareTitleIDs) == 0 {
+		return map[uint][]string{}, nil
+	}
+
+	stmt := `
+SELECT
+	st.id AS title_id,
+	sc.name AS software_category_name,
+	si.id AS installer_id
```

Comment:
Do we need installer_id here/in the VPP query/in the result struct? Doesn't seem like we use it?